server/pulp/server/db/manage.py

```diff
@@ -165,7 +165,16 @@ def ensure_database_indexes():
 
     # Load all the model classes that the server knows about and ensure their indexes as well
     plugin_manager = PluginManager()
-    for model_class in plugin_manager.unit_models.itervalues():
+    for unit_type, model_class in plugin_manager.unit_models.items():
+        unit_key_index = {'fields': model_class.unit_key_fields, 'unique': True}
+        for index in model_class._meta['indexes']:
+            if unit_key_index == index:
```

Comment:
I haven't tested it, but does this work as expected on Python 2.6 also? Somehow I remember dictionary equality comparison being a Python 2.7 thing.